src/plugin/xyz/xyzproviderimport.js

```diff
@@ -0,0 +1,206 @@
+goog.module('plugin.xyz.XYZImport');
+goog.module.declareLegacyNamespace();
+
+const {getRandomString, numerateCompare} = goog.require('goog.string');
+const ConfigProvider = goog.require('plugin.config.Provider');
+const {ROOT} = goog.require('os');
+const ConfigDescriptor = goog.require('os.data.ConfigDescriptor');
+const {createFromOptions} = goog.require('os.layer');
+const {getProjections} = goog.require('os.proj');
+const Module = goog.require('os.ui.Module');
+const {ID_DELIMITER} = goog.require('os.ui.data.BaseProvider');
+const SingleUrlProviderImportCtrl = goog.require('os.ui.SingleUrlProviderImportCtrl');
+const XYZProviderHelpUI = goog.require('plugin.xyz.XYZProviderHelpUI');
+
+const IDataDescriptor = goog.requireType('os.data.IDataDescriptor');
+
+
+/**
+ * The XYZ provider import directive
+ *
+ * @return {angular.Directive}
+ */
+const directive = () => {
+  return {
+    restrict: 'E',
+    replace: true,
+    templateUrl: ROOT + 'views/plugin/xyz/xyzsingleurl.html',
+    controller: Controller,
+    controllerAs: 'ctrl'
+  };
+};
+
+
+/**
+ * The element tag for the directive.
+ * @type {string}
+ */
+const directiveTag = 'xyzprovider';
+
+
+/**
+ * Add the directive to the module
+ */
+Module.directive(directiveTag, [directive]);
+
+
+/**
+ * Controller for the XYZ provider import dialog
+ * @unrestricted
+ */
+class Controller extends SingleUrlProviderImportCtrl {
+  /**
+   * Constructor.
+   * @param {!angular.Scope} $scope
+   * @param {!angular.JQLite} $element
+   * @ngInject
+   */
+  constructor($scope, $element) {
+    super($scope, $element);
+
+    /**
+     * @type {string}
+     */
+    this['helpUi'] = XYZProviderHelpUI.directiveTag;
+
+    // initialize units from settings
+    var projections = getProjections(true);
+    projections.sort(function(a, b) {
+      return numerateCompare(
+          /** @type {string} */ (a['code']),
+          /** @type {string} */ (b['code']));
+    });
+
+    this['projections'] = projections;
+
+    $scope['typeName'] = 'XYZ Provider';
+    $scope['urlExample'] = 'https://example.com/{z}/{x}/{y}';
+    $scope['config']['id'] = 'xyz';
+    $scope['config']['label'] = this.getLabel() || '';
+    $scope['config']['title'] = this.scope['config']['label'];
+    $scope['config']['type'] = 'XYZ';
+    $scope['config']['layerType'] = 'XYZ';
+    $scope['config']['url'] = '';
+    $scope['config']['minZoom'] = 1;
```

Comment:
I would suggest using `os.map.MIN_ZOOM` and `os.map.MAX_ZOOM` for the initial values here.